node/src/components/chain_synchronizer/operations.rs

```diff
@@ -904,6 +904,17 @@ pub(super) async fn run_chain_sync_task(
 
     verify_trusted_block_header(&chain_sync_context)?;
 
+    let lowest_contiguous_block_height = if config.sync_to_genesis() {
+        // If backfilling to genesis is moved to a background task, we should change this to
+        // `trusted_block_header.height()` and update storage as we backfill.
+        0
+    } else {
+        trusted_block_header.height()
+    };
+    effect_builder
+        .update_lowest_contiguous_block_height_in_storage(lowest_contiguous_block_height)
+        .await;
```

Comment:
In the case of `sync_to_genesis()` we'll persist `0` here even before we start syncing. Now, if the sync to genesis takes, let's say, a few hours, we'll pretend that we have all blocks, but we're actually still fetching them.

This could be misleading unless `joiner` doesn't expose the RPC API to be queried.

Reply:
Yes, the joiner has no RPC server, so not a problem.